extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1052,9 +1052,8 @@ export class Project implements ISqlProject {
 		// again when they convert to an SDK-style project
 		if (this.isSdkStyleProject) {
 			// update sqlproj if a node was deleted and load files and folders again
-			if (deleted) {
```

Comment:
removing this check because we still need to reload the project's files for the delete scenario so that a Build Remove entry doesn't get added for the files in the folder getting deleted.